datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -216,10 +216,9 @@ impl LogicalPlanBuilder {
         if values.is_empty() {
             return plan_err!("Values list cannot be empty");
         }
+
+        // values list can have no columns, see: https://github.com/apache/datafusion/pull/12339
         let n_cols = values[0].len();
-        if n_cols == 0 {
-            return plan_err!("Values list cannot be zero length");
-        }
```

Comment:
I use `LogicalPlanBuilder::values` to recompute the schema for `LogicalPlan::Values`
causing this line to fail the test from #12339.

Since we allow values without columns, I think this line should be removed.